src/VisualStudio/Core/Def/ServicesVSResources.resx

```diff
@@ -699,4 +699,7 @@ Additional information: {1}</value>
   <data name="Prefer_predefined_type" xml:space="preserve">
     <value>Prefer predefined type</value>
   </data>
+  <data name="Connection_to_remote_host_has_been_lost_some_features_might_stop_working_or_start_working_in_proc_do_you_want_to_recover_remote_host" xml:space="preserve">
+    <value>Connection to remote host has been lost. some features might stop working or start working in proc. do you want to recover remote host?</value>
```

Comment:
>also, I think user should be in the position where they decide whether they want to run things out of proc or not 

We need to meet over this.  But until we make the decisoin that this is something we want to do, i do not want us checking in the code that puts this information and decision in front of the user.  It would be akin to us crashing on some thread and then asking the user "running code asynchronously failed, do you want us to run it synchronously".  It's not something they have the context to make a decision on, nor can they understand the implication of their choice.  Finally, it's not something i believe we have the bandwidth to test or support.  Just getting in Out-Of-Proc is going to be a substantial cost, and we should be trying to minimize that to make sure that what we get in is as high quality as possible.